### ISSUE TITLE
Remove `uniqueVersion`?

### ISSUE THREAD

**Issue by stephanh** [CommBank/uniform#76]
Currently Uniform generates unique version numbers for us by including the setting `uniqueVersion` in our version.sbt file. However, in our CI build we have the CI generate the version number at the beginning and overwrite the version.sbt file. We do this so that the version number is the same during the whole build and survives several runs of sbt. It is also ensures that we do it the same way for non sbt builds.

The only place where `uniqueVersion` is actually used then is locally where it can be a bit of a pain when you publishing fat jars or dependencies for downstream testing.

Do we still want to keep `uniqueVersion` or should we get rid of it?

**Comment by toddmowen** [comment]
:+1: Really?? In that case get rid of it.

Is it possible to make local builds even more explicit, e.g. appending `-SNAPSHOT`?

**Comment by vineethvarghese** [comment]
@toddmowen Don't local uncommit build already do that?

**Comment by vineethvarghese** [comment]
@stephanh This is news :+1:

**Comment by stephanh** [comment]
@toddmowen yeah we could just replace it with something that always appends `-SNAPSHOT` to the version number.

**Comment by toddmowen** [comment]
Are we sure that _none_ of the CI builds depend on `uniqueVersionSettings`? I don't think we can know without a thorough audit (or just waiting for things to break).

I remember somebody (@jberthold?) mentioning that some artifacts on artifactory had `-SNAPSHOT` in their version, which must have been generated by the Uniform.

I say: go ahead and remove `uniqueVersionSettings`, but make sure people keep an eye out for unexpected results. Consistency is good, so this may actually be a good way to flush out any such "non-standard" builds that exist.

**Comment by rowandavies** [comment]
@stephanh Shall we close this?  Or leave it open until we actually remove the deprecated `uniqueVersionSettings` ?

**Comment by stephanh** [comment]
Close this on we have done it

**Comment by rowandavies** [comment]
The `uniform` project itself still uses a copy of `uniqueVersionSettings` that is in it's `project/version.scala` but is out of date (in fact hardly updated since the initial commit) .  Is there any reason not to just replace this with a copy of `localVersionSettings` ?
